### PR TITLE
feat: use iana media type vocabulary for format mapping, if eu format vocabulary does not match

### PR DESCRIPTION
**Pull request by sarusarah**

### CODE REVIEW

File: ckanext/dcatapchharvest/profiles.py [hunk 1]
```diff
@@ -902,6 +904,17 @@ def graph_from_dataset(self, dataset_dict, dataset_ref):  # noqa
...
+            if format_uri is None and resource_dict.get('media_type'):
+                for key, value in valid_media_types.items():
+                    if resource_dict.get('media_type') == key:
+                        media_type_uri = URIRef(value)
+                        g.add((
+                            distribution,
+                            DCT['format'],
+                            media_type_uri
+                        ))
+
```
Comment (sarusarah) [inline ckanext/dcatapchharvest/profiles.py:917]: Further question: should the URI of the IANA media type also be set for the DCT['media_type']?
And could be the case exist, that the IANA media type format is getting by resource_dict.get('format')?
In both cases the logic needs to be adjusted.

Comment (bellisk) [reply ckanext/dcatapchharvest/profiles.py:917]: >  should the URI of the IANA media type also be set for the DCT['media_type']?

Yes

> could be the case exist, that the IANA media type format is getting by resource_dict.get('format')?

Not sure what you mean here?

Comment (sarusarah) [reply ckanext/dcatapchharvest/profiles.py:954]: Added uses cases and comments for them. Can you have a look?

File: ckanext/dcatapchharvest/profiles.py [hunk 2]
```diff
@@ -903,8 +903,20 @@ def graph_from_dataset(self, dataset_dict, dataset_ref):  # noqa
                             DCT['format'],
                             format_uri
                         ))
+            # Media Type: Set Format value if format matches EU vocabulary
+            # and media type is not set
+            if format_uri and resource_dict.get('media_type') is None:
+                g.add((
+                    distribution,
+                    DCT['media_type'],
```
Comment (bellisk) [inline ckanext/dcatapchharvest/profiles.py:911]:
```suggestion
                    DCAT.mediaType,
```

File: ckanext/dcatapchharvest/profiles.py [hunk 3]
```diff
@@ -914,6 +926,31 @@ def graph_from_dataset(self, dataset_dict, dataset_ref):  # noqa
                             DCT['format'],
                             media_type_uri
                         ))
+                        g.add((
+                            distribution,
+                            DCT['media_type'],
```
Comment (bellisk) [inline ckanext/dcatapchharvest/profiles.py:931]:
```suggestion
                            DCAT.mediaType,
```

File: ckanext/dcatapchharvest/profiles.py [hunk 4]
```diff
@@ -914,6 +926,31 @@ def graph_from_dataset(self, dataset_dict, dataset_ref):  # noqa
...
+                        if resource_dict.get('format') == key:
+                            media_type_uri_by_format = URIRef(value)
+                            g.add((
+                                distribution,
+                                DCT['format'],
+                                media_type_uri_by_format
+                            ))
+                            g.add((
+                                distribution,
+                                DCT['media_type'],
```
Comment (bellisk) [inline ckanext/dcatapchharvest/profiles.py:951]:
```suggestion
                                DCAT.mediaType,
```

**Pull request merged**